File: 2025-1/sesion-de-clase-18/SoftProgNet/SoftProgPersistencia/DAOImpl/Logistica/Ventas/OrdenVentaDAOImpl.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.Common;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using PUCP.Edu.Pe.SoftProg.Modelo.Clientes;
using PUCP.Edu.Pe.SoftProg.Modelo.Logistica.Almacen;
using PUCP.Edu.Pe.SoftProg.Modelo.Logistica.Ventas;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAO;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAO.Logistica.Ventas;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.Clientes;
using PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.RRHH;

namespace PUCP.Edu.Pe.SoftProg.Persistencia.DAOImpl.Logistica.Ventas {
    public class OrdenVentaDAOImpl : BaseDAOImpl<OrdenVenta>, IOrdenVentaDAO {
        protected override DbCommand CommandoInsertar(DbConnection conn, OrdenVenta ordenVenta) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "insertarOrdenVenta";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_idCliente", DbType.Int32, ordenVenta.Cliente.Id);
            this.AgregarParametroEntrada(cmd, "@p_idEmpleado", DbType.Int32, DBNull.Value);
            this.AgregarParametroEntrada(cmd, "@p_total", DbType.Double, ordenVenta.Total);
            this.AgregarParametroEntrada(cmd, "@p_activo", DbType.Boolean, ordenVenta.IsActive);
            this.AgregarParametroSalida(cmd, "@p_id", DbType.Int32);

            return cmd;
        }

        protected override DbCommand CommandoInsertar(DbConnection conn, DbTransaction transaccion, OrdenVenta ordenVenta) {
            DbCommand cmd = this.CommandoInsertar(conn, ordenVenta);
            cmd.Transaction = transaccion;

            return cmd;
        }

        protected override DbCommand CommandoModificar(DbConnection conn, OrdenVenta ordenVenta) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "modificarOrdenVenta";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_idCliente", DbType.Int32, ordenVenta.Cliente.Id);
            this.AgregarParametroEntrada(cmd, "@p_idEmpleado", DbType.String, ordenVenta.Empleado.Id);
            this.AgregarParametroEntrada(cmd, "@p_total", DbType.String, ordenVenta.Total);
            this.AgregarParametroEntrada(cmd, "@p_activo", DbType.Boolean, ordenVenta.IsActive);
            this.AgregarParametroEntrada(cmd, "@p_id", DbType.Int32, ordenVenta.Id);

            return cmd;
        }

        protected override DbCommand CommandoEliminar(DbConnection conn, int id) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "eliminarOrdenVenta";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_id", DbType.Int32, id);

            return cmd;
        }

        protected override DbCommand CommandoBuscar(DbConnection conn, int id) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "buscarOrdenVentaPorId";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;

            this.AgregarParametroEntrada(cmd, "@p_id", DbType.Int32, id);

            return cmd;
        }

        protected override DbCommand CommandoListar(DbConnection conn) {
            DbCommand cmd = conn.CreateCommand();
            cmd.CommandText = "listarOrdenesVenta";
            cmd.CommandType = System.Data.CommandType.StoredProcedure;
            return cmd;
        }

        protected override OrdenVenta mapearModelo(DbDataReader reader) {
            return new OrdenVenta {
                Id = Convert.ToInt32(reader["id"]),
                Cliente = new ClienteDAOImpl().Buscar(Convert.ToInt32(reader["idCliente"])), 
                Empleado = new EmpleadoDAOImpl().Buscar(Convert.ToInt32(reader["idEmpleado"])), 
                Total = Convert.ToInt32(reader["total"]),
                IsActive = Convert.ToBoolean(reader["activo"])
            };
        }
    }
}

File: 2025-1/sesion-de-clase-18/SoftProg/SoftProgPersistencia/src/main/java/pe/edu/pucp/inf30/softprog/daoimpl/logistica/ventas/OrdenVentaDAOImpl.java
package pe.edu.pucp.inf30.softprog.daoimpl.logistica.ventas;

import java.sql.CallableStatement;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.Types;
import pe.edu.pucp.inf30.softprog.dao.logistica.ventas.IOrdenVentaDAO;
import pe.edu.pucp.inf30.softprog.daoimpl.BaseDAOImpl;
import pe.edu.pucp.inf30.softprog.daoimpl.clientes.ClienteDAOImpl;
import pe.edu.pucp.inf30.softprog.model.logistica.ventas.OrdenVenta;

/**
 *
 * @author eric
 */
public class OrdenVentaDAOImpl extends BaseDAOImpl<OrdenVenta> implements IOrdenVentaDAO {
    @Override
    protected CallableStatement comandoInsertar(Connection conn, OrdenVenta modelo) throws SQLException {
        String sql = "{CALL insertarOrdenVenta(?, ?, ?, ?, ?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt("p_idCliente", modelo.getCliente().getId());
        cmd.setNull("p_idEmpleado", Types.INTEGER);
        cmd.setDouble("p_total", modelo.getTotal());
        cmd.setBoolean("p_activo", modelo.isActivo());
        cmd.registerOutParameter("p_id", Types.INTEGER);
        return cmd;
    }

    @Override
    protected CallableStatement comandoModificar(Connection conn, OrdenVenta modelo) throws SQLException {
        String sql = "{CALL modificarOrdenVenta(?, ?, ?, ?, ?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt("p_idCliente", modelo.getCliente().getId());
        cmd.setNull("p_idEmpleado", Types.INTEGER);
        cmd.setDouble("p_total", modelo.getTotal());
        cmd.setBoolean("p_activo", modelo.isActivo());
        cmd.setInt("p_id", modelo.getId());
        return cmd;
    }

    @Override
    protected CallableStatement comandoEliminar(Connection conn, int id) throws SQLException {
        String sql = "{CALL eliminarOrdenVenta(?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt(1, id);
        return cmd;
    }

    @Override
    protected CallableStatement comandoBuscar(Connection conn, int id) throws SQLException {
        String sql = "{CALL buscarOrdenVentaPorId(?)}";
        CallableStatement cmd = conn.prepareCall(sql);
        cmd.setInt(1, id);
        return cmd;
    }

    @Override
    protected CallableStatement comandoListar(Connection conn) throws SQLException {
        String sql = "{CALL listarOrdenesVenta()}";
        CallableStatement cmd = conn.prepareCall(sql);
        return cmd;
    }

    @Override
    protected OrdenVenta mapearModelo(ResultSet rs) throws SQLException {
        OrdenVenta ordenVenta = new OrdenVenta();
        ordenVenta.setId(rs.getInt("id"));
        ordenVenta.setCliente(new ClienteDAOImpl().buscar(rs.getInt("idCliente")));
        ordenVenta.setTotal(rs.getDouble("total"));
        ordenVenta.setActivo(rs.getBoolean("activo"));
        return ordenVenta;
    }
}